src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -5,11 +5,25 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System.Net.Http.Headers
 {
+    /// <summary>Key/value pairs of headers. The value is either a raw <see cref="string"/> or a <see cref="HttpHeaders.HeaderStoreItemInfo"/>.</summary>
+    internal struct HeaderEntry
```

Comment:
Should we just use `KeyValuePair<HeaderDescriptor, object>` instead of defining our own struct?

Reply:
We need ref access to the fields, whereas KeyValuePair is readonly.

Reply:
Ahhh, interesting. Let's add a comment then.

Reply:
How about just using a tuple `(HeaderDescriptor Key, object Value)`?

Reply:
I suppose that would work if we prefer it over an explicit struct. Personally, I find `HeaderEntry[]` easier to read than `(HeaderDescriptor Key, object Value)[]`.